pkg/controllers/nodepool/hash/controller.go

```diff
@@ -69,3 +79,37 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) operatorcontr
 		WithOptions(controller.Options{MaxConcurrentReconciles: 10}),
 	)
 }
+
+func (c *Controller) updateNodeClaimHash(ctx context.Context, np *v1beta1.NodePool) error {
+	ncList := &v1beta1.NodeClaimList{}
+	if err := c.kubeClient.List(ctx, ncList, client.MatchingLabels(map[string]string{v1beta1.NodePoolLabelKey: np.Name})); err != nil {
+		return err
+	}
+
+	errs := make([]error, len(ncList.Items))
+	for i := range ncList.Items {
+		stored := ncList.Items[i].DeepCopy()
+
+		if ncList.Items[i].Annotations[v1beta1.NodePoolHashVersionAnnotationKey] != v1beta1.NodePoolHashVersion {
+			ncList.Items[i].Annotations = lo.Assign(ncList.Items[i].Annotations, map[string]string{
+				v1beta1.NodePoolHashVersionAnnotationKey: v1beta1.NodePoolHashVersion,
+			})
+
+			// Any NodeClaim that is already drifted will remain drifted if the karpenter.sh/nodepool-hash-version doesn't match
+			// Since the hashing mechanism has changed we will not be able to determine if the drifted status of the node has changed
+			if ncList.Items[i].StatusConditions().GetCondition(v1beta1.Drifted) == nil {
+				ncList.Items[i].Annotations = lo.Assign(ncList.Items[i].Annotations, map[string]string{
+					v1beta1.NodePoolHashAnnotationKey: np.Hash(),
+				})
+			}
```

Comment:
Don't we want to check this first, and not update the hash version of the nodeclaim if it's drifted? If we just update the nodepool hash version in the first iteration, we'll just update it on the second iteration.

Reply:
We will not update the `nodepool-hash` on the NodeClaim, if the `nodepool-hash-version`  match the NodePool

Reply:
I could go either way on this. I don't think that it tangibly matters one way or the other here. If you don't update the hash but update the version, this just means nodes that are already drifted are still going to stay drifted because their hashes _still_ won't match the NodePool's hash.

Reply:
Yup! That was my thinking